### PR TITLE
Card: Improve accessibility by moving `overlay` slot outside of `<a>` tag

### PR DESCRIPTION
**Pull request by michael-iden**
**This change:** (check at least one)

- [ ] Adds a new feature
- [X] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [X] No - Although I think its pretty close to a breaking change

**Is the:** (complete all)

- [X] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [ ] Test suite(s) passing?
- [ ] Code coverage maximal?
- [ ] Changeset added?
- [X] Component status page up to date?

**What does this change address?**
Fixes a problem where the button inside of the `overlay` slot would not be picked up by the browser when tabbing through clickable elements. I believe this is because the button within the `overlay` was slotted inside of an `anchor` tag

**How does this change work?**
Move overlay slot outside of the link tag

**Additional context**
I am very nervous about this change. This component is far too complex and takes in too many variable elements through slots. Even with tests and the sandbox I have little confidence that it's not actually going to break the UX in at least some browsers.

Also I am struggling to get the hooks to run properly with the shimmed yarn version. My global yarn in `4.0.1` in the node envs 18+ managed by `asdf`. I might look to update the application setup to a later yarn if we can get a moment (#635).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -147,5 +139,5 @@
       "path": "packages/pharos/lib/index.js"
     }
   ],
-  "packageManager": "yarn@1.22.19"
+  "packageManager": "yarn@4.0.2"
```
Comment (michael-iden) [inline package.json:142]: This isn't right

File: packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx [hunk 2]
```diff
@@ -390,6 +390,53 @@ export const WithActionButtonSlot = {
     `,
 };
 
+export const WithOverlayButtonSlot = {
```
Comment (daneah) [inline packages/pharos/src/components/image-card/pharos-image-card.wc.stories.jsx:393]: Can you make sure to duplicate this in the React stories as well?

File: package.json (hunk 1 above)
Comment (daneah) [reply package.json:142]: Let's see if we can revert all the Yarn-related items for the moment.

**Comment by brentswisher** [comment]
@michael-iden Wondering what the status of this is, I see it has been open a few months. Is there work needed here? Should it be closed as no longer needed?

**Comment by michael-iden** [comment]
@brentswisher this is still an issue but was unfortunately deprioritized and I haven't gotten back around to updating things. I'll try to get an estimate on when I might be able to circle back on this fix and I'll send you a thread directly where we were tangentially talking about this problem.

**Comment by brentswisher** [comment]
@michael-iden No worries at all, appreciate the work in this, just trying to do a little GitHub cleanup where I can! 😄

**Pull request closed without merging**